test/gui/shared/scripts/helpers/SetupClientHelper.py

```diff
@@ -67,17 +70,88 @@ def getResourcePath(context, resource, user=None):
 
 def startClient(context):
     squish.startApplication(
-        "owncloud -s"
+        "owncloud OWNCLOUD_CORE_DUMP=1 -s"
         + " --logfile "
         + context.userData['clientConfigFile']
         + " --logdebug"
         + " --logflush"
         + " --confdir "
         + confdir
     )
+    squish.installEventHandler("Crash", "crashHandler")
+
     squish.snooze(1)
 
 
+def crashHandler():
+    cfg = ConfigParser()
+
+    # Select appropriate AUT binary with respect to the test run environment
+    if environ.get('CI'):
+        try:
+            cfg.read('/drone/src/test/gui/drone/server.ini')
+            owncloudBinary = cfg.get('General', 'AUT/owncloud')
+            owncloudBinary = owncloudBinary.replace('"', '') + '/owncloud'
+        except Exception as err:
+            test.log("the binary path can not be read")
+            test.log(err)
+    else:
+        try:
+            from pathlib import Path
+
+            HOME = str(Path.home())
+            cfg.read(HOME + '/.squish/ver1/server.ini')
+            owncloudBinary = cfg.get('General', 'AUT/owncloud')
+            owncloudBinary = owncloudBinary.replace('"', '') + '/owncloud'
+        except Exception as err:
+            test.log("the binary path can not be read")
+            test.log(err)
+
+    # GUI test stacktrace on crash
+    test.log("Started printing the GUI test stacktrace of crash")
+    test.log(
+        "###############################################################################################################"
+    )
+    test.log("%s" % test.stackTrace())
+    test.log(
+        "###############################################################################################################"
+    )
+    test.log("Finished printing GUI test stacktrace")
+
+    # The core dump is generated in the folder /var/lib/apport/coredump/
+    test.log("Started printing the AUT stacktrace of crash")
+    test.log(
+        "###############################################################################################################"
+    )
+    ls = subprocess.run(
+        ['ls', '-t', '/var/lib/apport/coredump'], stdout=subprocess.PIPE
```

Comment:
save that path in a constant, so that if it changes it only has to be changed in one place